public/app/core/components/QueryOperationRow/QueryOperationRow.test.tsx

```diff
@@ -1,56 +1,42 @@
-import { mount, shallow } from 'enzyme';
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
 import React from 'react';
-import { act } from 'react-dom/test-utils';
 
-import { QueryOperationRow } from './QueryOperationRow';
+import { QueryOperationRow, QueryOperationRowProps } from './QueryOperationRow';
+
+const setup = (propOverrides?: Partial<QueryOperationRowProps>) => {
+  const props: QueryOperationRowProps = {
+    title: 'test-title',
+    headerElement: '',
+    index: 0,
+    id: 'test-id',
+    children: <div>children</div>,
+    ...propOverrides,
+  };
+  return render(<QueryOperationRow {...props}></QueryOperationRow>);
+};
 
 describe('QueryOperationRow', () => {
-  it('renders', () => {
-    expect(() =>
-      shallow(
-        <QueryOperationRow id="test-id" index={0}>
-          <div>Test</div>
-        </QueryOperationRow>
-      )
-    ).not.toThrow();
+  it('renders without exploding', () => {
+    expect(() => setup()).not.toThrow();
   });
 
-  describe('callbacks', () => {
-    it('should not call onOpen when component is shallowed', async () => {
-      const onOpenSpy = jest.fn();
-      // @ts-ignore strict null error, you shouldn't use promise like approach with act but I don't know what the intention is here
-      await act(async () => {
-        shallow(
-          <QueryOperationRow onOpen={onOpenSpy} id="test-id" index={0}>
-            <div>Test</div>
-          </QueryOperationRow>
-        );
-      });
-      expect(onOpenSpy).not.toBeCalled();
-    });
+  it('renders the component content', () => {
+    setup();
+    expect(screen.getByText(/^test-title$/)).toBeInTheDocument();
+  });
 
+  describe('callbacks', () => {
     it('should call onOpen when row is opened and onClose when row is collapsed', async () => {
       const onOpenSpy = jest.fn();
       const onCloseSpy = jest.fn();
-      const wrapper = mount(
-        <QueryOperationRow title="title" onOpen={onOpenSpy} onClose={onCloseSpy} isOpen={false} id="test-id" index={0}>
-          <div>Test</div>
-        </QueryOperationRow>
-      );
-      const titleEl = wrapper.find({ 'aria-label': 'Query operation row title' });
-      expect(titleEl).toHaveLength(1);
-
-      // @ts-ignore strict null error, you shouldn't use promise like approach with act but I don't know what the intention is here
-      await act(async () => {
-        // open
-        titleEl.first().simulate('click');
-      });
-
-      // @ts-ignore strict null error, you shouldn't use promise like approach with act but I don't know what the intention is here
-      await act(async () => {
-        // close
-        titleEl.first().simulate('click');
-      });
+      setup({ isOpen: false, onOpen: onOpenSpy, onClose: onCloseSpy });
+
+      const queryRow = screen.getByText(/^test-title$/);
+      expect(queryRow).toBeInTheDocument();
+
+      await userEvent.click(queryRow);
+      await userEvent.click(queryRow);
```

Comment:
Document that the first click is for opening and second for closing (as done in the original code) would be helpful here.